src/Tickets/Commerce/Gateways/PayPal/REST/Order_Endpoint.php

```diff
@@ -158,6 +155,107 @@ public function handle_create_order( WP_REST_Request $request ) {
 		return new WP_REST_Response( $response );
 	}
 
+	/**
+	 * Retrieves the unit data for an item in the cart.
+	 *
+	 * By default, the item type will be considered a 'ticket' if not specified.
+	 * This method handles different item types with a switch case, providing custom logic
+	 * for 'ticket' and a default behavior for other item types.
+	 * An overarching filter allows for customization of the final returned data.
+	 *
+	 * @since TBD
+	 *
+	 * @param array   $item The cart item for which to retrieve unit data.
+	 * @param WP_Post $order The order from the items in the cart.
+	 *
+	 * @return array The structured data for the item, including 'name', 'unit_amount', 'quantity', 'item_total', and
+	 *     'sku'.
+	 */
+	public function get_unit_data( array $item, WP_Post $order ) {
+		if ( ! $order->ID ) {
+			return [];
+		}
+
+		$type = $item['type'] ?? 'ticket';
+
+		switch ( $type ) {
+			case 'ticket':
+				$unit_data = $this->get_unit_data_for_ticket( $item, $order );
+				break;
+
+			default:
+				/**
+				 * Filters the unit data for custom item types in the cart.
+				 *
+				 * This filter allows external developers to generate and customize the unit data
+				 * for items in the cart based on the item type (other than 'ticket').
+				 *
+				 * The filter name is dynamic and uses the item type (`$type`) to provide flexibility for
+				 * different item categories.
+				 *
+				 * Example: If `$type` is 'fee', the filter will be `tec_commerce_get_unit_data_fee`.
+				 *
+				 * @since TBD
+				 *
+				 * @param array   $item   The cart item for which the unit data is being generated.
+				 * @param WP_Post $order  The current order object.
+				 *
+				 * @return array The unit data for the item.
+				 */
+				$unit_data = apply_filters( "tec_tickets_commerce_paypal_order_get_unit_data_{$type}", $item, $order );
+				break;
+		}
+
+		/**
+		 * Filters the unit data for an item in REST context.
+		 *
+		 * @since TBD
+		 *
+		 * @param array   $unit_data The structured data for the item.
+		 * @param array   $item      The order item for which the unit data is being generated.
+		 * @param WP_Post $order     The current order object.
+		 */
+		return apply_filters( 'tec_tickets_commerce_paypal_order_get_unit_data', $unit_data, $item, $order );
+	}
+
+	/**
+	 * Retrieves the default unit data for a ticket in the cart.
+	 *
+	 * This method is used when the item type is 'ticket', and it structures the data
+	 * for a ticket item, including details such as name, price, quantity, and SKU.
+	 *
+	 * @since TBD
+	 *
+	 * @param array   $item The cart item (representing the ticket).
+	 * @param WP_Post $order The order from the items in the cart.
+	 *
+	 * @return array The structured data for the ticket item.
```

Comment:
```suggestion
	 * @return array<string,mixed> The structured data for the ticket item.
```

Better if we could do the whole structure like [this](https://github.com/the-events-calendar/event-tickets/blob/7ca697bedf568ba7de101311a9dd8fc98b42d402/src/Tickets/Seating/Tables/Sessions.php#L283-L287).